From 404c2dc2c9e81d775bf10f0d23d8fdbc64aaa6ff Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 13 Aug 2019 15:53:30 +0300
Subject: [PATCH] [SW-0] pcie: gaudi update skip logic

We align all HL-2000 products to behave like HL-205 i.e. active low

Note: Inform all HL200 users to update their DIP switch

Change-Id: Ied2878adba9682568537f072411aab70df342146
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 319331c..c760a0d 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -297,8 +297,8 @@
 
 u32 hl_skip_pcie_init(void)
 {
-	/* Use DumpSel[11] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
+	/* Use DumpSel[11] to skip PCIe initialization (active low) */
+	return !regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & BIT(19);
 }
 
 u64 hl_pll_get_cpu_cycles_per_usec(void)
